### PR TITLE
recommed content api for karmayogi users

### PR DESCRIPTION
**Pull request by shankaragoudab**

### CODE REVIEW

File: src/main/java/org/sunbird/common/util/CbExtServerProperties.java [hunk 1]
```diff
@@ -610,6 +610,27 @@ public void setRedisDataPort(String redisDataPort) {
...
+	private String notifyServiceHost;
+
+	@Value("${notify.service.path}")
+	private String notifyServicePath;
+
+	@Value("${lms.service.host}")
+	private String lmsServiceHost;
+
+	@Value("${lms.user.search}")
+	private String lmsUserSearchEndPoint;
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/common/util/CbExtServerProperties.java]: use the existing properties

File: src/main/java/org/sunbird/common/util/CbExtServerProperties.java [hunk 2]
```diff
@@ -610,6 +610,27 @@ public void setRedisDataPort(String redisDataPort) {
...
+
+	@Value("${recommend.content.notification.template}")
+	private String recommendContentTemplate;
+
+	@Value("${notification.support.mail}")
+	private String supportEmail;
+
+	@Value("${notify.service.path.async}")
+	private String notifyServicePathAsync;
+
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/common/util/CbExtServerProperties.java:624]: use the existing properties

File: src/main/java/org/sunbird/core/config/PropertiesConfig.java [hunk 3]
```diff
@@ -129,4 +130,11 @@ public String getNotificationAsyncPath() {
     public void setNotificationAsyncPath(String notificationAsyncPath) {
         this.notificationAsyncPath = notificationAsyncPath;
     }
+    public String getRecommendContentTemplate() {
+        return recommendContentTemplate;
+    }
+
+    public void setRecommendContentTemplate(String recommendContentTemplate) {
+        this.recommendContentTemplate = recommendContentTemplate;
+    }
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/core/config/PropertiesConfig.java]: why this changes are required ?

File: src/main/java/org/sunbird/user/service/UserUtilityService.java [hunk 4]
```diff
@@ -49,4 +50,5 @@ public void getUserDetailsFromDB(List<String> userIds, List<String> fields,
 	Map<String, Map<String, Object>> getUserDetailsFromES(List<String> userIdList, List<String> userFields) throws IOException;
 
 	Map<String, Object> getUsersDataFromLookup(String email, String authToken);
+	SBApiResponse getRecommendation(String authUserToken, Map<String, Object> orgRequest);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/user/service/UserUtilityService.java]: use proper method name... we are not trying to get something...

File: src/main/java/org/sunbird/user/service/UserUtilityServiceImpl.java [hunk 5]
```diff
@@ -770,4 +765,208 @@ public Map<String, Object> getUsersDataFromLookup(String email, String authToken
...
+				mailNotificationDetails.put(Constants.LINK, link.toString());
+				mailNotificationDetails.put(Constants.COURSE_POSTER_IMAGE_URL, requestData.get(Constants.COURSE_POSTER_IMAGE_URL));
+				mailNotificationDetails.put(Constants.COURSE_PROVIDER, requestData.get(Constants.COURSE_PROVIDER));
+				mailNotificationDetails.put(Constants.USER_ID, userId);
+				mailNotificationDetails.put(Constants.USER, name);
+				sendNotificationToRecipients(mailNotificationDetails);
+			}
+		} catch (Exception e) {
+			String errMsg = "Error while performing operation." + e.getMessage();
+			logger.error(errMsg + e);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/user/service/UserUtilityServiceImpl.java]: it suppose to be logger.error(errMsg, e);

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/org/sunbird/user/service/UserUtilityService.java (hunk 4 above)
Comment (shankaragoudab) [reply src/main/java/org/sunbird/user/service/UserUtilityService.java]: updated

File: src/main/java/org/sunbird/common/util/CbExtServerProperties.java (hunk 1 above)
Comment (shankaragoudab) [reply src/main/java/org/sunbird/common/util/CbExtServerProperties.java]: updated

File: src/main/java/org/sunbird/core/config/PropertiesConfig.java (hunk 3 above)
Comment (shankaragoudab) [reply src/main/java/org/sunbird/core/config/PropertiesConfig.java]: updated

File: src/main/java/org/sunbird/common/util/CbExtServerProperties.java (hunk 2 above)
Comment (shankaragoudab) [reply src/main/java/org/sunbird/common/util/CbExtServerProperties.java:624]: updated
